projects/packages/publicize/src/rest-endpoints/class-connections-controller.php

```diff
@@ -0,0 +1,303 @@
+	 */
+	protected static function get_all_connections( $run_tests = false ) {
+		/**
+		 * Publicize instance.
+		 *
+		 * @var \Automattic\Jetpack\Publicize\Publicize $publicize
+		 */
+		global $publicize;
+
+		$items = array();
+
+		$test_results = $run_tests ? self::get_connections_test_status() : array();
+
+		foreach ( (array) $publicize->get_services( 'connected' ) as $service_name => $connections ) {
+			foreach ( $connections as $connection ) {
+
+				$connection_id = $publicize->get_connection_id( $connection );
+
+				$connection_meta = $publicize->get_connection_meta( $connection );
+				$connection_data = $connection_meta['connection_data'];
+
+				$items[] = array(
+					'connection_id'        => $connection_id,
+					'display_name'         => $publicize->get_display_name( $service_name, $connection ),
+					'external_handle'      => $publicize->get_external_handle( $service_name, $connection ),
+					'external_id'          => $connection_meta['external_id'] ?? '',
+					'profile_link'         => $publicize->get_profile_link( $service_name, $connection ),
+					'profile_picture'      => $publicize->get_profile_picture( $connection ),
+					'service_label'        => Publicize::get_service_label( $service_name ),
+					'service_name'         => $service_name,
+					'shared'               => ! $connection_data['user_id'],
+					'status'               => $test_results[ $connection_id ] ?? 'ok',
+					'user_id'              => (int) $connection_data['user_id'],
+
+					// Deprecated fields.
+					'id'                   => (string) $publicize->get_connection_unique_id( $connection ),
+					'username'             => $publicize->get_username( $service_name, $connection ),
+					'profile_display_name' => ! empty( $connection_meta['profile_display_name'] ) ? $connection_meta['profile_display_name'] : '',
+					// phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual -- We expect an integer, but do loose comparison below in case some other type is stored.
+					'global'               => 0 == $connection_data['user_id'],
+
+				);
+			}
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Get a list of publicize connections.
+	 *
+	 * @param bool $run_tests Whether to run tests on the connections.
+	 *
+	 * @return array
+	 */
+	public static function get_connections( $run_tests = false ) {
+		if ( self::is_wpcom() ) {
+			return self::get_all_connections( $run_tests );
+		}
+
+		$site_id = Manager::get_site_id( true );
+		if ( ! $site_id ) {
+			return array();
+		}
+
+		$path = add_query_arg(
+			array( 'test_connections' => $run_tests ),
+			sprintf( '/sites/%d/publicize/connections', $site_id )
+		);
+
+		$response = Client::wpcom_json_api_request_as_user( $path, 'v2', array( 'method' => 'GET' ) );
+
+		if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			// TODO log error.
+			return array();
+		}
+
+		$body = wp_remote_retrieve_body( $response );
+
+		$items = json_decode( $body, true );
```

Comment:
Should we check here to make sure it's an array?

Reply:
I think the emptiness check below in return statement is sufficient.